tests/orm/data/test_singlefile.py

```diff
@@ -105,6 +106,45 @@ def test_construct_from_string(self):
         self.assertEqual(content_stored, content_original)
         self.assertEqual(node.list_object_names(), [SinglefileData.DEFAULT_FILENAME])
 
+    def test_construct_with_path(self):
+        """Test constructing an instance from a pathlib.Path."""
+        content_original = 'please report to the ministry of silly walks'
+
+        with tempfile.NamedTemporaryFile(mode='w+') as handle:
+            filepath = pathlib.Path(handle.name).resolve()
+            basename = os.path.basename(filepath)
+            handle.write(content_original)
+            handle.flush()
+            node = SinglefileData(file=filepath)
+
+        uuid = node.uuid
+
+        with node.open() as handle:
+            content_written = handle.read()
+
+        self.assertEqual(node.list_object_names(), [basename])
+        self.assertEqual(content_written, content_original)
+
+        node.store()
+
+        with node.open() as handle:
+            content_stored = handle.read()
+
+        self.assertEqual(content_stored, content_original)
+        self.assertEqual(node.list_object_names(), [basename])
+
+        node_loaded = load_node(uuid)
+        self.assertTrue(isinstance(node_loaded, SinglefileData))
+
+        with node.open() as handle:
+            content_loaded = handle.read()
+
+        self.assertEqual(content_loaded, content_original)
+        self.assertEqual(node_loaded.list_object_names(), [basename])
+
+        with node_loaded.open() as handle:
+            self.assertEqual(handle.read(), content_original)
```

Comment:
Is this copied from another test?

Do we need to test this again?

Reply:
Yeah that was copied from the surrounding tests - there was actually quite a bit of duplication in these tests. Couldn't help myself and refactored that into two fixtures.